src/components/app/ListOfPublishedProfiles.js

```diff
@@ -72,38 +77,225 @@ function _formatRange(range: StartEndRange): string {
+    }
+
+    // Let's focus the delete button after dismissing the dialog, but _only_ if
+    // the focus was part of the dialog before.
+    // Note: we might need to get a more precise ref to the right
+    // buttonWithPanel wrapper when we'll have more buttons.
+    const deleteButton = this._domDeleteButtonRef.current;
+    const actionButtons = this._actionButtonsRef.current;
+    try {
+      if (
+        deleteButton &&
+        actionButtons &&
+        actionButtons.matches(':focus-within')
+      ) {
+        deleteButton.focus();
+      }
+    } catch (e) {
+      // This browser doesn't support :focus-within (especially JSDOM), let's
+      // degrade gracefully by not refocusing the delete button.
+    }
+  };
+
+  onOpenConfirmDialog = () => {
+    this.setState({ confirmDialogIsOpen: true });
+  };
+
+  preventClick(e) {
+    e.preventDefault();
+  }
+
+  _renderPossibleErrorMessage() {
+    const { error } = this.state;
+    if (!error) {
+      return null;
+    }
+
+    return (
+      <p className="publishedProfilesDeleteError">
+        An error happened while deleting this profile.{' '}
+        <a href="#" title={error.message} onClick={this.preventClick}>
+          Hover to know more.
+        </a>
+      </p>
+    );
+  }
+
+  render() {
+    const { profileData, nowTimestamp, withActionButtons } = this.props;
+    const { confirmDialogIsOpen, status } = this.state;
+
+    if (status === 'deleted') {
+      return null;
+    }
+
+    let { urlPath } = profileData;
+    if (!urlPath.startsWith('/')) {
+      urlPath = '/' + urlPath;
+    }
+    const location = `${window.location.origin}/${urlPath}`;
+    const slicedProfileToken = profileData.profileToken.slice(0, 6);
+    const profileName = profileData.name
+      ? profileData.name
+      : `Profile #${slicedProfileToken}`;
+    const smallProfileName = profileData.name
+      ? profileData.name
+      : '#' + slicedProfileToken;
+
+    return (
+      <li
+        className={classNames('publishedProfilesListItem', {
+          publishedProfilesListItem_ConfirmDialogIsOpen: confirmDialogIsOpen,
+        })}
+      >
+        <a
+          className="publishedProfilesLink"
+          href={location}
+          title={`Click here to load profile ${smallProfileName}`}
+        >
+          <div className="publishedProfilesDate">
+            {_formatDate(profileData.publishedDate, nowTimestamp)}
+          </div>
+          <div className="publishedProfilesInfo">
+            <div className="publishedProfilesName">
+              <strong>{profileName}</strong> (
+              {_formatRange(profileData.publishedRange)})
+            </div>
+            <ProfileMetaInfoSummary meta={profileData.meta} />
+          </div>
+        </a>
+        {withActionButtons ? (
+          <div
+            className="publishedProfilesActionButtons"
+            ref={this._actionButtonsRef}
+          >
+            {profileData.jwtToken ? (
+              <ButtonWithPanel
```

Comment:
I'll probably extract this whole `ButtonWithPanel` with the deletion behavior to a separate component in the next patch.